runtime/lua/vim/shared.lua

```diff
@@ -544,7 +542,8 @@ end
 ---@return string %-escaped pattern string
 function vim.pesc(s)
   vim.validate({ s = { s, 's' } })
-  return s:gsub('[%(%)%.%%%+%-%*%?%[%]%^%$]', '%%%1')
+  local result = s:gsub('[%(%)%.%%%+%-%*%?%[%]%^%$]', '%%%1')
+  return result
```

Comment:
this change doesn't look necessary

Reply:
The original code returns two values instead of one.

Reply:
Ah, then it helps to assign to `_` to make it clear what's intended.
```suggestion
  local result, _ = s:gsub('[%(%)%.%%%+%-%*%?%[%]%^%$]', '%%%1')
  return result
```

or just wrap in parens:

    return (s:gsub('[%(%)%.%%%+%-%*%?%[%]%^%$]', '%%%1'))

Reply:
Also needs a test https://github.com/justinmk/neovim/blob/373fef462bc3d23f8b38b6fb3ab942bbf1b283c0/test/functional/lua/vim_spec.lua#L430